Prevent wrapping null storage

PVince81 opened the pull request:
Can happen when trying to instantiate external storages that have
incomplete config, where the constructor throws an exception (the
exception is caught in createStorage())

Please review @icewind1991 @Xenopathic 

If this approach is fine I'll add a unit test.

I suspect this could happen on stable7 if a third party app uses storage wrappers.

PVince81 commented:
Fixes https://github.com/owncloud/core/issues/13715

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6a8117d1-de96-4cc7-b80f-22b4a7c54658) was created.

PVince81 commented:
also @MorrisJobke

PVince81 commented:
- [ ] add unit test

LukasReschke commented on lib/private/files/mount/mountpoint.php:
```diff
@@ -42,6 +42,14 @@ class MountPoint implements IMountPoint {
+	 *
+	 * @var bool
+	 */
+	private $invalidStorage;
```
Just for clearness initialize with false?

LukasReschke replied:
(or true – whatever suits better ;))

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d362961d-7f12-4a8f-a36e-312040469fd8) was created.

icewind1991 commented:
Looks good :+1:

PVince81 commented:
Ok thanks. I'll add a unit test then.
*checks tests folder*
NOOOOOOOOOOOOOOOOooo this class has no unit test

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6a8117d1-de96-4cc7-b80f-22b4a7c54658) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d362961d-7f12-4a8f-a36e-312040469fd8) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/69e377e8-d66a-4835-aee4-30c56c7f868d) was created.

PVince81 commented:
I moved setting `$invalidStorage` into `createStorage`. There's another case where the class doesn't exist, so it will also set it to true.

And added unit tests as well.

Please review :smile:

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/69e377e8-d66a-4835-aee4-30c56c7f868d) completed: **6 new issues, 3 updated code elements**

MorrisJobke commented:
Fixes the issue :+1:
